projects/igniteui-angular/src/lib/drop-down/drop-down.component.ts

```diff
@@ -137,6 +107,73 @@ export class IgxDropDownComponent extends IgxDropDownBase implements IDropDownBa
     @Output()
     public onClosed = new EventEmitter<void>();
 
+    /**
+     * Gets/sets whether items take focus. Disabled by default.
+     * When enabled, drop down items gain tab index and are focused when active -
+     * this includes activating the selected item when opening the drop down and moving with keyboard navigation.
+     *
+     * Note: Keep that focus shift in mind when using the igxDropDownItemNavigation directive
+     * and ensure it's placed either on each focusable item or a common ancestor to allow it to handle keyboard events.
+     *
+     * ```typescript
+     * // get
+     * let dropDownAllowsItemFocus = this.dropdown.allowItemsFocus;
+     * ```
+     *
+     * ```html
+     * <!--set-->
+     * <igx-drop-down [allowItemsFocus]='true'></igx-drop-down>
+     * ```
+     */
+    @Input()
+    public allowItemsFocus = false;
+
+    /**
+     * @hidden @internal
+     */
+    public get focusedItem(): IgxDropDownItemBase {
+        if (this.virtDir) {
+            return this._focusedIndex !== -1 ? this.children.find(e => e.index === this._focusedIndex) : null;
+        }
+        return this._focusedItem;
+    }
+
+    public set focusedItem(value: IgxDropDownItemBase) {
+        this._focusedItem = value;
+        this.selection.clear(`${this.id}-active`);
+        if (value) {
```

Comment:
For now, index access should be limited to either inside the item (i.e. save value directly to the service) or behind virtDir check here:
```suggestion
        if (value && this.virtDir) {
```
Otherwise it'll log the auto-index warning on non-virtualized DD without index bound.